### PR TITLE
Fix deployment script to replace hostnames and IPs automatically

### PR DESCRIPTION
**Pull request by muhammadimranfarooqi**
@vkuznet ,

I have fixed script that will automatically update hostnames/IPs in ingresses for deployment script. For the frontend deployment, the script will be used as:

CMSWEB_HOSTNAME=cmsweb-test4.cern.ch ./scripts/deploy.sh create frontend /k8s/path/config-test4 /certificates/certificatesfrontend-test4 hmac 

For the backend deployment, it will be used as:
CMSWEB_HOSTNAME_FRONTEND=cmsweb-test4.cern.ch CMSWEB_HOSTNAME=cmsweb-srv4.cern.ch ./scripts/deploy.sh create services /k8s/path/config-test4 /certificates/certificatesbackend-srv4 hmac

### CODE REVIEW

**Comment by muhammadimranfarooqi** [comment]
@vkuznet 

I have done changes in cmsweb-deployment.md file for documentation as well as per your suggestions. Could you please review. 

Best
Muhammad Imran

File: kubernetes/cmsweb/scripts/deploy.sh [hunk 1]
```diff
@@ -462,16 +463,17 @@ deploy_ingress()
...
     tmpDir=/tmp/$USER/ingress
     mkdir -p $tmpDir
     for ing in $cmsweb_ing; do
         cp ingress/${ing}.yaml $tmpDir
         cat ingress/${ing}.yaml | \
             awk '{if($1=="nginx.ingress.kubernetes.io/whitelist-source-range:") {print "    nginx.ingress.kubernetes.io/whitelist-source-range: "ips""} else print $0}' ips=$ips > \
-            $tmpDir/${ing}.yaml
-        echo "deploy ingress: $tmpDir/${ing}.yaml"
+            $tmpDir/${ing}.tmp.yaml
+	cat $tmpDir/${ing}.tmp.yaml | awk '{if($2=="host:") {print "  - host : "hostname""} else print $0}' hostname=$cmsweb_hostname >  $tmpDir/${ing}.yaml
```
Comment (vkuznet) [inline kubernetes/cmsweb/scripts/deploy.sh]: Here I suggested to do it in single pass:
```
cat ingress/${ing}.yaml | \
    awk '{if($1=="nginx.ingress.kubernetes.io/whitelist-source-range:") {print "    nginx.ingress.kubernetes.io/whitelist-source-range: "ips""} else print $0}' ips=$ips | \
    awk '{if($2=="host:") {print "  - host : "hostname""} else print $0}' hostname=$cmsweb_hostname \
    > $tmpDir/${ing}.yaml
```

So, do not create intermediate tmp.yaml file and make it as one pass through UNIX pipes. Please test and adjust PR.

**Comment by muhammadimranfarooqi** [comment]
@vkuznet ,

Sorry, I made a mistake here. I committed in the wrong version. Please dont merge this file. I'll send again.

**Comment by muhammadimranfarooqi** [comment]
@vkuznet 

Could you please review PR now.  I have corrected mistake and tested locally. All services are working. 

Best
Muhammad

**Pull request merged**